src/filters.rs

```diff
@@ -166,6 +166,8 @@ impl RowFilter {
                 match self {
                     Self::None => unreachable!(),
                     Self::Sub => {
+                        // Clippy seems to be triggering a false positive here
+                        #[expect(clippy::needless_range_loop)]
                         for j in 0..color_bytes {
                             pixels[i][j] = pixels[prev][j];
                         }
```

Comment:
I've taken this lint as an excuse to slightly optimize the function in [this commit](https://github.com/oxipng/oxipng/commit/2a3aa839282fa5a9796c4a7b9be393be0eef0973), so this lint expectation can now be removed.

Reply:
It indeed translates to more instructions overall, but we now avoid doing useless work on the `prev == i` case, and the asserts to eliminate bound checks help a little with branch prediction. I'd find it completely reasonable to just silence the lint instead as you did, though :joy: